pypit/artrace.py

```diff
@@ -1613,102 +1651,1643 @@ def slit_trace_qa(slf, frame, ltrace, rtrace, extslit, desc="",
+                    # Traces can disappear and then the crude trace can wind up hitting a neighbor
+                    # Therefore, take only the continuous good piece from the starting point
+                    ybad_xerr = np.where(~goodx[:,kk])[0]
+                    # Lower point
+                    ylow = ybad_xerr < yrow
+                    if np.any(ylow):
+                        y0 = np.max(ybad_xerr[ylow])+1  # Avoid the bad one
+                    else:
+                        y0 = 0
+                    # Upper
+                    yhi = ybad_xerr > yrow
+                    if np.any(yhi):
+                        y1 = np.min(ybad_xerr[yhi])
+                    else:
+                        y1 = edgearr.shape[0]
+                    new_yval = np.arange(y0,y1).astype(int)  # Yes, this is necessary;  slicing fails..
+                    #
+                    new_edgarr[new_yval, xvals[new_yval]] = eval
+                    #new_edgarr[yval, xvals[yval]] = eval
+                # Flag
+                tc_dict[side]['flags'][uni_e == eval] = 1
+                # Save new_xval
+                tc_dict[side]['xval'][str(eval)] = new_xval
+                # Zero out edgearr
+                edgearr[edgearr==eval] = 0
+            # Next pass
+            niter += 1
+
+    # Reset edgearr values to run sequentially and update the dict
+    for side in ['left', 'right']:
+        if np.any(tc_dict[side]['flags'] == -1):
+            # Loop on good edges
+            gde = np.where(tc_dict[side]['flags'] == 1)[0]
+            for ss,igde in enumerate(gde):
+                if side == 'left':
+                    newval = -1*ednum - ss
+                else:
+                    newval = ednum + ss
+                oldval = tc_dict[side]['uni_idx'][igde]
+                pix = new_edgarr == oldval
+                new_edgarr[pix] = newval
+                # Reset the dict too..
+                if newval == 0:
+                    debugger.set_trace()
+                tc_dict[side]['xval'][str(newval)] = tc_dict[side]['xval'].pop(str(oldval))
+    print(tc_dict['left']['xval'])
+    print(tc_dict['right']['xval'])
+    # Return
+    return new_edgarr, tc_dict.copy()
+
+
+def mslit_sync(edgearr, tc_dict, ednum, insert_buff=5, add_left_edge_slit=True):
+    """ Method to synchronize the slit edges
+    Adds in extra edges according to a few criteria
+
+    Developed for ARMLSD alone
+
+    Parameters
+    ----------
+    edgearr : ndarray
+    tc_dict : dict
+       For book-keeping
+    ednum : int
+    insert_buff : int, optional
+       Offset from existing edge for any edge added in
+    add_left_edge_slit : bool, optional
+       Allow the method to add in a left slit at the edge of the detector
+
+    Returns
+    -------
+    new_edgearr : ndarray
+      Updated edgearr
+    """
+
+    # Internal method (for convenience)
+    def add_edge(ref_slit, insert_offset, earr, t_dict, final_left, final_right, left=True):
```

Comment:
Anytime we do this (functions inside functions), we should think about whether we should turn the whole operation into a class object.  It depends on how this is convenient.

Reply:
For example, what would happen if you thought of the `edgearr` as an object that you're constructing?  All the functions currently called `edgearr_*` would likely be an internal function for the class, and the final object would keep the final slit edges internally that could be used by or passed to other functions.